test/dns-test-srv/main.go

```diff
@@ -53,6 +98,20 @@ func dnsHandler(w dns.ResponseWriter, r *dns.Msg) {
 			record.Preference = 10
 
 			m.Answer = append(m.Answer, record)
+		case dns.TypeTXT:
+			ts.mu.RLock()
+			if value, present := ts.txtRecords[q.Name]; present {
```

Comment:
Just for practice, let's put these variables on their own line only lock around the setting around them